### PR TITLE
N Knight Kill in Java

### LINKED ISSUE #5835: N Knight Kill in Java
## 💥 Proposal

I would like to add the N Knight KIll problem in java. It is a backtracking problem in which there are N knights that need to be placed on an N*N board in such a manner so that no two knights can kill each other.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

### PR DESCRIPTION
**Pull request by AnshikaAgrawal5501**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

N Knight Kill is a backtracking problem in which there are N knights that are needed to be placed on an N*N board in such a manner so that no two knights can kill each other. It will display all the possible positions of N Knights.
[Link](https://replit.com/@Anshikaagrawal5/N-Knight-Kill)
In the sample output, I have written only 1 possibility since there are many boards possible for a particular N.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes : #5835

### CODE REVIEW

File: Java/other/NKnightKill.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+Enter no. of Knights / board size - 3
+
+Output (Many boards are possible therefore showing only one):
+
+Possibility 1 - 
+X   X   X
+-   -   -
+-   -   -
+
+*/
```
Comment (bhav09) [inline Java/other/NKnightKill.java]: Add a new line at EOF

File: Java/other/NKnightKill.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Space Complexity : O(N*N)
+
+Input:
+
+Enter no. of Knights / board size - 3
+
+Output (Many boards are possible therefore showing only one):
+
+Possibility 1 - 
+X   X   X
```
Comment (bhav09) [inline Java/other/NKnightKill.java:100]: Can you please mention what does this 'X' mean?

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed. Please remove redundant 'new lines'.

File: Java/other/NKnightKill.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	
+	private static void nKnight(char[][] board, int i, int j, int n) {
+
+		if (n==0) {
+			System.out.println("Possibility "+count+" - ");
+			display (board);
+			count++;
+			System.out.println();
+			return;
```
Comment (AnshikaAgrawal5501) [inline Java/other/NKnightKill.java:50]: Hey @bhav09, you asked me to put a new line at EOF. Please see here, I have already used an Sopln statement to make sure that not only all the boards get printed with a line gap but also for the very last board that will get printed there will definitely be a new line at the end.

Please have a look at this. There is a line gap at the end.
Untitled

If I am misunderstanding your point then please elaborate a bit.
Thanks.

**Comment by bhav09** [comment]
I have edited, confirm the commit from your side.  Can you explain what does 'X' mean in the comments in the code?
Thanks :)

File: Java/other/NKnightKill.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/**
+It is a backtracking problem in which there are N knights 
+that need to be placed on an N*N board in such a manner so 
+that no two knights can kill each other.
+The code will display all the possible positions of Knights
+where - will represent a blank box and 
+X will represent a Knight in a box.
+*/
```
Comment (AnshikaAgrawal5501) [inline Java/other/NKnightKill.java:8]: I have just added the meaning of X here, in the description of the program. Please review.
And also thanks for your help.

**Comment by AnshikaAgrawal5501** [comment]
Hey @bhav09, Please add the level label.
Thanks.

**Pull request merged**